functional-tests/support/selectors/index.ts

```diff
@@ -25,14 +27,19 @@ const pageMappings = [
 	},
 ];
 
-export const getSelector = (selectorName) => {
-	const browserPath = new URL(browser.getUrl()).pathname,
-		pageMapping = pageMappings.find(({ path }) =>
-			typeof path === "string" ? path === browserPath : path.test(browserPath)
-		),
-		pageSelectors = pageMapping && pageMapping.selectors,
-		selectors = Object.assign({}, commonSelectors, pageSelectors),
-		selector = selectors[selectorName];
+export const getSelector = async (
+	selectorName: SelectorName
+): Promise<string> => {
+	const browserPath = new URL(await browser.getUrl()).pathname,
+		pageMapping = pageMappings.find(({ path }) => path === browserPath);
+
+	const pageSelectors = pageMapping && pageMapping.selectors;
+
+	const selectors = { ...commonSelectors, ...pageSelectors } as Record<
```

Comment:
Just curious what this `as` record is.... Not seen this syntax before.

Reply:
`as` is TypeScript casting, and we're casting to `Record` which is kind of like key/value pairs or a dictionary, but it's generic so you can specify the type of the key and value. You could do it like:

```
type Thing = {
    [key: SelectorName]: string;
}
const selectors = { ...commonSelectors, ...pageSelectors } as Thing;
```

But I think the record syntax is more elegant and clearer of what's happening

Reply:
Ah I see, thanks for the explanation. :-)